android/src/main/java/com/dylanvann/fastimage/FastImageViewConverter.java

```diff
@@ -74,11 +77,25 @@ static Headers getHeaders(ReadableMap source) {
             }
 
             headers = builder.build();
-        }
+        } catch (Exception e) {
+            // If headers are passed as an array, convert them to a map
+            ReadableArray headersArray = source.getArray("headers");
+            LazyHeaders.Builder builder = new LazyHeaders.Builder();
+
+            for (int i = 0; i < headersArray.size(); i++) {
+                ReadableMap headerMap = headersArray.getMap(i);
+                String header = headerMap.getString("name");
+                String value = headerMap.getString("value");
+                builder.addHeader(header, value);
+            }
 
-        return headers;
+            headers = builder.build();
```

Comment:
This should be the default and the only logic instead of handling it inside the catch block. We can remove the try block code since it is obselete.

Reply:
@Anb98, let me know if you have any questions on the above comment.

Reply:
I updated the code as you requested, let me know your thoughts